docs/conf.py

```diff
@@ -296,3 +302,104 @@ def setup(app):
 if read_the_docs_build:
 
     subprocess.call('cd ../minorminer/docs/; make cpp', shell=True)
+
+# Link to GitHub source
+github_map = {'dwavebinarycsp': ['dwavebinarycsp',],
+              'cloud': ['dwave-cloud-client',],
+              'dimod':  ['dimod',],
+              'dwave_networkx': ['dwave-networkx',],
+              'greedy': ['dwave-greedy',],
+              'hybrid': ['dwave-hybrid',],
+              'inspector': ['dwave-inspector',],
+              'minorminer': ['minorminer',],
+              'neal': ['dwave-neal',],
+              'penaltymodel': [{'cache': ['penaltymodel_cache',],
+                               'core': ['penaltymodel_core',],
+                               'lp': ['penaltymodel_lp',],
+                               'maxgap': ['penaltymodel_maxgap',],
+                               'mip': ['penaltymodel_mip',]}],
+              'system': ['dwave-system',],
+              'embedding': ['dwave-system',],
+              'tabu': ['dwave-tabu',]}
+
+os.system('cd .. ; python setup.py egg_info; cd docs')
+with open('../dwave_ocean_sdk.egg-info/requires.txt', 'r') as requires_txt:
+    repo_versions = requires_txt.readlines()
+    for line in repo_versions:
+        if line.split('==')[0] == 'penaltymodel':
+            github_map['penaltymodel'][0]['core'].append(line.split('==')[1].rstrip())
+        for gh, repo in github_map.items():
+            if gh=='penaltymodel':
+               for gh, repo in repo[0].items():
+                  if repo[0].replace('_', '-') in line:
+                     github_map['penaltymodel'][0][gh].append(line.split('==')[1].rstrip())
+            else: # non penalty model
+               if repo[0] in line:
+                  github_map[gh].append(line.split('==')[1].rstrip())
```

Comment:
Can be simplified a lot by using `pkg_resources` to fetch project's requirements:
```python
import pkg_resources
reqs = pkg_resources.get_distribution('dwave-ocean-sdk').requires(extras=['all'])
pkgs = [pkg_resources.get_distribution(req) for req in reqs]
versions = {pkg.project_name: pkg.version for pkg in pkgs}
```
This produces:
```
>>> versions
{'dimod': '0.9.13',
 'dwave-cloud-client': '0.8.4',
 'dwave-greedy': '0.1.2',
 'dwave-hybrid': '0.6.1',
 'dwave-inspector': '0.2.5',
 ...}
```

Note that we have to use `all` extras to include both `penaltymodel-maxgap` and `penaltymodel-mip`. Also, for this to work, versions of packages installed in the current venv have to match those required by `install_requires`. Currently, RtD is set-up to install docs deps with `pip install -r docs/requirements.txt` and package deps via `python setup.py install`. This might result in a version mismatch, and we should install all deps using the same method, e.g. pip. To do this, I'd recommend (and we should do it regardless) to configure RtD build through `.readthedocs.yml`, where we would have a section like this:
```
python:
   version: 3.7
   install:
      - method: pip
        path: .
      - requirements: docs/requirements.txt
```

Reply:
Replaced `python setup.py egg_info` with `pkg_resources`. Still need to update `.readthedocs.yml`

Reply:
The recommendation to use a YAML file is implemented with https://github.com/dwavesystems/dwave-ocean-sdk/pull/123